### PR TITLE
Fix(web-react): Polymorph Link component types #DS-1060

### PR DESCRIPTION
**Pull request by pavelklibani**
### Issue reference

---

### Before submitting the PR, please make sure you do the following

- [ ] Read the [Contributing Guidelines](https://github.com/lmc-eu/spirit-design-system/blob/main/CONTRIBUTING.md).
- [ ] Follow the [PR Title/Commit Message Convention](https://github.com/lmc-eu/spirit-design-system/blob/main/CONTRIBUTING.md#commit-conventions).
- [ ] Provide a description in this PR that addresses **what** the PR is solving, or reference the issue that it solves (e.g. `fixes #123`).
- [ ] Ideally, include relevant tests that fail without this PR but pass with it.

### CODE REVIEW

File: packages/web-react/src/components/Link/Link.tsx [hunk 1]
```diff
@@ -1,18 +1,22 @@
...
 import { useStyleProps } from '../../hooks';
-import { SpiritLinkProps } from '../../types';
+import { SpiritLinkProps, PolymorphicRef } from '../../types';
 import { useLinkStyleProps } from './useLinkStyleProps';
 
 const defaultProps = {
   color: 'primary',
 };
 
+/**
```
Comment (literat) [inline packages/web-react/src/components/Link/Link.tsx:11]: IMHO this should be near the type definition, where it is, so I suggest removing it from here. I think we will use this type in multiple places soon and copy of this comment will be unnecessary.

File: packages/web-react/src/components/Link/README.md [hunk 2]
```diff
@@ -34,4 +35,29 @@ const CustomLink = (props: SpiritLinkProps): JSX.Element => {
...
+```jsx
+import { forwardRef } from 'react
+import { Link } from '@lmc-eu/spirit-web-react';
+import { PolymorphicRef } from '@lmc-eu/spirit-web-react/types';
+
+type LinkProps<T extends ElementType = 'button'> = SpiritLinkProps<T, 'tertiary'>;
+
+const CustomLinkRoot = <T extends ElementType = 'button'>(
+  props: LinkProps<T>,
+  ref: PolymorphicRef<T>
```
Comment (literat) [inline packages/web-react/src/components/Link/README.md:51]:
```suggestion
  ref: PolymorphicRef<T> // <-- Type `ref` prop with the `PolymorphicRef` here
```

File: packages/web-react/src/types/shared/element.ts [hunk 3]
```diff
@@ -19,6 +19,11 @@ export type SpiritPolymorphicElementPropsWithoutRef<E extends ElementType, P> =
   ComponentPropsWithoutRef<E>,
   keyof P
 >;
+/**
+ * For more information about `PolymorphicRef` type:
```
Comment (literat) [inline packages/web-react/src/types/shared/element.ts:23]:
```suggestion
 * Type React's forwarded ref with `PolymorphicRef` to allow generic `elementType` to be strongly typed, e.g. component allows switching of elements
```

**Review by literat — APPROVED** [review]
LGTM 👍

File: packages/web-react/src/components/Link/Link.tsx (hunk 1 above)
Comment (pavelklibani) [reply packages/web-react/src/components/Link/Link.tsx:11]: removed

File: packages/web-react/src/components/Link/README.md (hunk 2 above)
Comment (pavelklibani) [reply packages/web-react/src/components/Link/README.md:51]: comment added 👍

File: packages/web-react/src/types/shared/element.ts (hunk 3 above)
Comment (pavelklibani) [reply packages/web-react/src/types/shared/element.ts:23]: changed, thank you for better description ❤️

**Pull request merged**